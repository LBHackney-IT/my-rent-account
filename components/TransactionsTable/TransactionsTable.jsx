import React from "react";
import PropTypes from "prop-types";

import { getProperCurrency } from "utils/string";

const transactionsTable = ({ transactions }) => (
  <div style={{ overflow: "auto" }}>
    <table className="govuk-table">
      <thead className="govuk-table__head">
        <tr className="govuk-table__row">
          <th scope="col" className="govuk-table__header">
            Date
          </th>
          <th scope="col" className="govuk-table__header">
            Description
          </th>
          <th
            scope="col"
            className="govuk-table__header govuk-table__header--numeric"
          >
            Rent
          </th>
          <th
            scope="col"
            className="govuk-table__header govuk-table__header--numeric"
          >
            Paid In
          </th>
          <th
            scope="col"
            className="govuk-table__header govuk-table__header--numeric"
          >
            Balance
          </th>
        </tr>
      </thead>
      <tbody className="govuk-table__body">
        {transactions.map((transaction, i) => (
          <tr key={i} className="govuk-table__row">
            <td className="govuk-table__cell">
              {new Date(transaction.date).toLocaleDateString()}
            </td>
            <td className="govuk-table__cell">{transaction.description}</td>
            <td className="govuk-table__cell govuk-table__cell--numeric">
              {getProperCurrency(transaction.out)}
            </td>
            <td className="govuk-table__cell govuk-table__cell--numeric">
              {getProperCurrency(transaction.in)}
            </td>
            <td className="govuk-table__cell govuk-table__cell--numeric">
              {getProperCurrency(transaction.balance)}
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  </div>
);

transactionsTable.propTypes = {
  transactions: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      out: PropTypes.string,
      in: PropTypes.string,
      balance: PropTypes.string.isRequired,
    }).isRequired
  ).isRequired,
};

export default transactionsTable;
