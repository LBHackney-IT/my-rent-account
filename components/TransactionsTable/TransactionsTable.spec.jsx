import React from "react";
import { render } from "@testing-library/react";

import TransactionsTable from "./TransactionsTable";

describe("TransactionsTable component", () => {
  const props = {
    transactions: [
      {
        date: "2016-09-05T00:00:00",
        description: "Total Charge",
        valueIn: "",
        valueOut: "造0.00",
        balance: "造185.68",
      },
      {
        date: "2016-04-10T00:00:00",
        description: "Housing Benefit",
        valueIn: "造121.16",
        valueOut: "",
        balance: "造214.92",
      },
    ],
  };
  it("should render properly", () => {
    const { asFragment } = render(<TransactionsTable {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
