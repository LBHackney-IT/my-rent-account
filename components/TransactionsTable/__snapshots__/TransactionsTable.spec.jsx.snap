// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsTable component should render properly 1`] = `
<DocumentFragment>
  <div
    style="overflow: auto;"
  >
    <table
      class="govuk-table"
    >
      <thead
        class="govuk-table__head"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header"
            scope="col"
          >
            Date
          </th>
          <th
            class="govuk-table__header"
            scope="col"
          >
            Description
          </th>
          <th
            class="govuk-table__header govuk-table__header--numeric"
            scope="col"
          >
            Rent
          </th>
          <th
            class="govuk-table__header govuk-table__header--numeric"
            scope="col"
          >
            Paid in
          </th>
          <th
            class="govuk-table__header govuk-table__header--numeric"
            scope="col"
          >
            Balance
          </th>
        </tr>
      </thead>
      <tbody
        class="govuk-table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            05/09/2016
          </td>
          <td
            class="govuk-table__cell"
          >
            Total Charge
          </td>
          <td
            class="govuk-table__cell govuk-table__cell--numeric"
          >
            £0.00
          </td>
          <td
            class="govuk-table__cell govuk-table__cell--numeric"
          />
          <td
            class="govuk-table__cell govuk-table__cell--numeric"
          >
            £185.68
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <td
            class="govuk-table__cell"
          >
            10/04/2016
          </td>
          <td
            class="govuk-table__cell"
          >
            Housing Benefit
          </td>
          <td
            class="govuk-table__cell govuk-table__cell--numeric"
          />
          <td
            class="govuk-table__cell govuk-table__cell--numeric"
          >
            £121.16
          </td>
          <td
            class="govuk-table__cell govuk-table__cell--numeric"
          >
            £214.92
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
